### ISSUE TITLE
How to use elastixImageFilter.ExecuteInverse()?

### ISSUE THREAD

**Issue by Julie-Von** [SuperElastix/SimpleElastix#278]
I am new to SimpleElastix. I want to warp points from the moving image to fixed image, then I need the inverse transform. Following the instructions in this link https://simpleelastix.readthedocs.io/PointBasedRegistration.html, here are two ways: 
1) computed manually (see section 6.1.6 in the elastix manual), **but I could not find the corresponding operation in simpleElastix mapping to the -t0 command in elastix;**
2) via elastixImageFilter.ExecuteInverse(). **But an error raised:"AttributeError: 'ElastixImageFilter' object has no attribute 'ExecuteInverse'".**

Anyone who can help me on this?

Thank you very much!

**Comment by kaspermarstal** [comment]
That function was removed because it was not robust. 

Since then we have fixed a bug that might make this possible again. Will put it on my todo list.

**Comment by Julie-Von** [comment]
Many thanks.

Regarding the 1st method, is there any corresponding operation in simpleElastix mapping to the elastix -..... -t0 command in elastix? Because I could not use elastix, due to some installation problems.

**Comment by kaspermarstal** [comment]
Yes the transform parameter file given to elastixImageFilter.SetInitialTransformParameterFileName() is equivalent to -t0 on the command line.

**Comment by AnaisB2390** [comment]
Hello, I am also very new to Simple Elastix and I also need to warp points from the moving image to fixed image, then I need the inverse transform. 
I did not understood quit well the previous answers, can anyone show me an exemple of the command lies to use to execute this operation with Simple Elastix ? 

Many thanks !

**Comment by reynoldscem** [comment]
> Hello, I am also very new to Simple Elastix and I also need to warp points from the moving image to fixed image, then I need the inverse transform.
> I did not understood quit well the previous answers, can anyone show me an exemple of the command lies to use to execute this operation with Simple Elastix ?
> 
> Many thanks !

@AnaisB2390 Did you ever find a solution to this?

@kaspermarstal Is this still in the pipeline?

**Comment by gandreadis** [comment]
For anyone still looking for the workaround for this issue, I think this version works:

```
    elastixImageFilter = sitk.ElastixImageFilter()
    elastixImageFilter.SetInitialTransformParameterFileName('TransformParameters.0.txt')
    elastixImageFilter.SetFixedImage(fixed_image)
    elastixImageFilter.SetMovingImage(fixed_image)
    elastixImageFilter.LogToConsoleOff()
    rigid_pm = affine_parameter_map()
    elastixImageFilter.SetParameterMap(rigid_pm)
    elastixImageFilter.SetParameter('HowToCombineTransforms', 'Compose')
    elastixImageFilter.SetParameter('Metric', 'DisplacementMagnitudePenalty')

    elastixImageFilter.Execute()
    Tx = elastixImageFilter.GetTransformParameterMap()
    Tx[0]['InitialTransformParametersFileName'] = ('NoInitialTransform',)

    transformixImageFilter = sitk.TransformixImageFilter()
    transformixImageFilter.SetTransformParameterMap(Tx)
    transformixImageFilter.SetMovingImage(fixed_image)
    transformixImageFilter.SetFixedPointSetFileName(pts_filename)
    transformixImageFilter.Execute()
```

Important: This assumes that the previous transform (the one you want to invert) is in `TransformParameters.0.txt`.

**Comment by lucaslovercio** [comment]
Hi @gandreadis 
thanks for sharing the code. I adapted it to my problem but I had to increase the number of iterations to around 1500. In that way, it produces a transformation of the points. However, the points are not posed where they have to be in the transformed image. Did it work for you?
what does the function affine_parameter_map() return? it is like sitk.GetDefaultParameterMap("affine")?
thanks!

**Comment by gandreadis** [comment]
Hi @lucaslovercio, that's unfortunate to hear! Yes, `affine_parameter_map()` returns something like the default parameter map, but then a bit customized. These are the key params:

```
	p["Transform"] = ["AffineTransform"]
	p["Metric"] = ["AdvancedMattesMutualInformation"]
	p["NumberOfResolutions"] = ["4"]
	p["Registration"] = ["MultiResolutionRegistration"]
	p["AutomaticTransformInitialization"] =  ["true"]
	p["AutomaticTransformInitializationMethod"] = ["Origins"]
	p["AutomaticParameterEstimation"] = ["true"]
	p["CheckNumberOfSamples"] = ["true"]
	p["FixedImagePyramid"] = ["FixedSmoothingImagePyramid"]
	p["MovingImagePyramid"] = ["MovingSmoothingImagePyramid"]
	p["DefaultPixelValue"] = ["0"]
	p["Interpolator"] = ["LinearInterpolator"]
	p["FinalBSplineInterpolationOrder"] = ["1"]
	p["Resampler"] = ["DefaultResampler"]
	p["ResampleInterpolator"] = ["FinalBSplineInterpolator"]
	p["Optimizer"] = ["AdaptiveStochasticGradientDescent"]
	p["MaximumNumberOfIterations"] = ["2048"]
	p["ImageSampler"] = ["RandomCoordinate"]
	p["NumberOfSamplesForExactGradient"] = ["4096"]
	p["NumberOfSpatialSamples"] = ["4096"]
	p["MaximumNumberOfSamplingAttempts"] = ["8"]
	p["NewSamplesEveryIteration"] = ["true"]
```

For me, this approach seems to work so far, but then again, there might be some situations where it doesn't? I'd be curious to hear when/if you've found what the culprit is in your situation.
